src/Diagnostics/FxCop/System.Runtime.Analyzers/Core/Design/DefineAccessorsForAttributeArguments.cs

```diff
@@ -22,22 +22,44 @@ public abstract class DefineAccessorsForAttributeArgumentsAnalyzer : DiagnosticA
         internal const string AddAccessorCase = "AddAccessor";
         internal const string MakePublicCase = "MakePublic";
         internal const string RemoveSetterCase = "RemoveSetter";
-        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArguments), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
 
-        internal static DiagnosticDescriptor Rule = new DiagnosticDescriptor(RuleId,
-                                                                         s_localizableTitle,
-                                                                         "{0}",
-                                                                         DiagnosticCategory.Design,
-                                                                         DiagnosticSeverity.Warning,
-                                                                         isEnabledByDefault: false,
-                                                                         helpLinkUri: "http://msdn.microsoft.com/library/ms182136.aspx",
-                                                                         customTags: WellKnownDiagnosticTags.Telemetry);
+        private static readonly LocalizableString s_localizableTitle = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArguments), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+        private static readonly LocalizableString s_defaultRuleMessage = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArgumentsDefault), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+        private static readonly LocalizableString s_increaseVisiblityMessage = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArgumentsIncreaseVisibility), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+        private static readonly LocalizableString s_removeSetterMessage = new LocalizableResourceString(nameof(SystemRuntimeAnalyzersResources.DefineAccessorsForAttributeArgumentsRemoveSetter), SystemRuntimeAnalyzersResources.ResourceManager, typeof(SystemRuntimeAnalyzersResources));
+
+        internal static DiagnosticDescriptor DefaultRule = new DiagnosticDescriptor(RuleId,
```

Comment:
@mavasani @srivatsn @ManishJayaswal that is what we said when we were discussing on build/live de-duplication.

without tool/id uniqueness, there is no way to properly de-dup. if that is okay, duplicated id should be okay, otherwise, we need the uniqueness (tool, id) tuple.